src/InstrumentationEngine/ProfilerManager.cpp

```diff
@@ -2828,7 +2843,11 @@ HRESULT CProfilerManager::COMClassicVTableDestroyed(
     // Compiler complains that these callbacks taking void* parameters are ambiguous. Can't use variadic templates on this call.
     CComPtr<ICorProfilerCallback> pCallback;
 
-    shared_ptr<CProfilerCallbackHolder> pProfilerCallbackHolder = atomic_load(&m_profilerCallbackHolder);
+    CProfilerCallbackHolder* pProfilerCallbackHolder = static_cast<CProfilerCallbackHolder*>(InterlockedCompareExchangePointer(
+        (volatile PVOID*)&m_profilerCallbackHolder,
+        nullptr,
+        nullptr));
+
     if (pProfilerCallbackHolder != nullptr)
     {
         pCallback = (ICorProfilerCallback*)(m_profilerCallbackHolder->GetMemberForInterface(__uuidof(ICorProfilerCallback)));
```

Comment:
Same issue here: we should not be using the field directly

Reply:
Thanks for the catch